### PR TITLE
Stream Resolution Selection

### PR DESCRIPTION
**Pull request by jakermx**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the [**CONTRIBUTING**](Contributing.md) document.

### All Submissions:

* [ x] Have you followed the guidelines in our Contributing document?
* [x ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ x] Have you successfully ran tests with your changes locally?

## Screenshots (if appropriate):

[You can erase any parts of this template not applicable to your Pull Request.]
In order to limit the resolution of playback, I modified the proc that generates the MPD file that ISH uses, It select all available  resolutions below and inclusive the limit, if nothing found there is a fallback to the lowest available resolution.

I just add suport for SD, 640, 720o, 1080p, 4k....
I also added the option to select the minimun available resolution so it looks for sothing higher than 1 but the lowest available.... 

Hope helps
:)

### CODE REVIEW

File: resources/settings.xml [hunk 1]
```diff
@@ -1,132 +1,132 @@
...
-    <setting id="auto_skip_credits" type="bool" label="30079" default="false" visible="eq(-1,true)" subsetting="true"/>
-    <setting id="pause_on_skip" type="bool" label="30080" default="false" visible="eq(-1,true)" subsetting="true"/>
-    <setting id="UpNextNotifier_enabled" type="bool" label="30129" default="true"/>
-    <setting id="is_settings_upnext" type="action" label="30178" action="Addon.OpenSettings(service.upnext)" enable="System.HasAddon(service.upnext)" option="close" visible="eq(-1,true)" subsetting="true"/>
-    <setting id="ResumeManager_enabled" type="bool" label="30176" default="true"/>
-    <setting id="ResumeManager_dialog" type="bool" label="30177" default="true" visible="eq(-1,true)" subsetting="true"/>
-    <setting id="forced_subtitle_workaround" type="bool" label="30181" default="true" />
-  </category>
-  <category label="30023"><!--Expert-->
-    <setting label="30115" type="lsep"/>
```
Comment (Smeulf) [inline resources/settings.xml]: Shouldn't be removed IMO

File: resources/language/resource.language.en_gb/strings.po [hunk 2]
```diff
@@ -1,793 +1,825 @@
...
-msgid "\r\nThese files are used by the provider information scrapers to integrate movies and tv shows info, useful with bonus episodes or director's cut"
-msgstr ""
-
-msgctxt "#30193"
-msgid "Include all information in NFO files (use only with 'Local Information' scraper)"
-msgstr ""
+# Kodi Media Center language file
+# Addon Name: Netflix
+# Addon id: plugin.video.netflix
+# Addon Provider: libdev + jojo + asciidisco
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: Doesn't match last language commit https://github.com/CastagnaIT/plugin.video.netflix/commit/bdd11afdfd04713da390054787599dd44eeb45d5

File: resources/language/resource.language.en_gb/strings.po [hunk 3]
```diff
@@ -1,793 +1,825 @@
...
+msgctxt "#30171"
+msgid "Trending now"
+msgstr ""
+
+msgctxt "#30172"
+msgid "Popular on Netflix"
+msgstr ""
+
+msgctxt "#30173"
+msgid "NETFLIX ORIGINALS"
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: Doesn't match last language commit https://github.com/CastagnaIT/plugin.video.netflix/commit/bdd11afdfd04713da390054787599dd44eeb45d5

File: resources/language/resource.language.en_gb/strings.po [hunk 4]
```diff
@@ -1,793 +1,825 @@
...
+msgctxt "#30135"
+msgid "Cache cleared"
+msgstr ""
+
+msgctxt "#30136"
+msgid "Enable 1080p Unlock (not recommended for Android)"
+msgstr ""
+
+msgctxt "#30137"
+msgid "Enable VP9 profiles (not use with HEVC, disable if it causes artifacts)"
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: Doesn't match last language commit https://github.com/CastagnaIT/plugin.video.netflix/commit/bdd11afdfd04713da390054787599dd44eeb45d5

File: resources/language/resource.language.en_gb/strings.po [hunk 5]
```diff
@@ -1,793 +1,825 @@
...
+msgctxt "#30193"
+msgid "Include all information in NFO files (use only with 'Local Information' scraper)"
+msgstr ""
+
+msgctxt "#30194"
+msgid "Stream Type"
+msgstr ""
+
+msgctxt "#30195"
+msgid "Video Strem Type (Up to...)"
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: I would say "Limit video resolution to", but that's a matter of taste

File: resources/lib/services/msl/profiles.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 """MSL video profiles"""
 from __future__ import unicode_literals
-
+import xbmc
```
Comment (Smeulf) [inline resources/lib/services/msl/profiles.py]: Not sure it's used in the code

File: resources/language/resource.language.en_gb/strings.po [hunk 7]
```diff
@@ -1,793 +1,825 @@
...
+msgstr ""
+
+msgctxt "#30196"
+msgid "SD"
+msgstr ""
+
+msgctxt "#30197"
+msgid "640"
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: 640p ? Or SD 640p ?

File: resources/language/resource.language.en_gb/strings.po [hunk 8]
```diff
@@ -1,793 +1,825 @@
...
+msgctxt "#30194"
+msgid "Stream Type"
+msgstr ""
+
+msgctxt "#30195"
+msgid "Video Strem Type (Up to...)"
+msgstr ""
+
+msgctxt "#30196"
+msgid "SD"
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: SD 480p ?

File: resources/language/resource.language.en_gb/strings.po [hunk 9]
```diff
@@ -1,793 +1,825 @@
...
+msgstr ""
+
+msgctxt "#30197"
+msgid "640"
+msgstr ""
+
+msgctxt "#30198"
+msgid "720p"
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: HD 720p ?

File: resources/language/resource.language.en_gb/strings.po [hunk 10]
```diff
@@ -1,793 +1,825 @@
...
+msgstr ""
+
+msgctxt "#30199"
+msgid "1080p"
```
Comment (Smeulf) [inline resources/language/resource.language.en_gb/strings.po]: HD 1080p ?

**Comment by Smeulf** [comment]
Also, a question. What if I don't want to limit the resolution at all and have IE 8K whan avaliable without waiting for an update? How can I do? If I'm correct, your setting is mandatory, and ther's no way to avoid limiting...

Last but not least, for me, SD should limit to 480p, not 639.

File: resources/settings.xml [hunk 11]
```diff
@@ -1,132 +1,132 @@
...
+    <setting id="auto_skip_credits" type="bool" label="30079" default="false" visible="eq(-1,true)" subsetting="true"/>
+    <setting id="pause_on_skip" type="bool" label="30080" default="false" visible="eq(-1,true)" subsetting="true"/>
+    <setting id="UpNextNotifier_enabled" type="bool" label="30129" default="true"/>
+    <setting id="is_settings_upnext" type="action" label="30178" action="Addon.OpenSettings(service.upnext)" enable="System.HasAddon(service.upnext)" option="close" visible="eq(-1,true)" subsetting="true"/>
+    <setting id="ResumeManager_enabled" type="bool" label="30176" default="true"/>
+    <setting id="ResumeManager_dialog" type="bool" label="30177" default="true" visible="eq(-1,true)" subsetting="true"/>
+    <setting id="forced_subtitle_workaround" type="bool" label="30181" default="true" />
+  </category>
+  <category label="30023"><!--Expert-->
+    <setting id="StreamType" type="enum" label="30195" lvalues="30196|30197|30198|30199|30200|30201" default="0"/>
```
Comment (Smeulf) [inline resources/settings.xml]: Default is 0, so SD. People may complain about quality.
It should be 4, unless the setting is not mandatory.

File: resources/language/resource.language.en_gb/strings.po (hunk 5 above)
Comment (jakermx) [reply resources/language/resource.language.en_gb/strings.po]: Agreed

File: resources/lib/services/msl/profiles.py (hunk 6 above)
Comment (jakermx) [reply resources/lib/services/msl/profiles.py]: You are right, it sould not be there, I used for debugging...

File: resources/language/resource.language.en_gb/strings.po (hunk 7 above)
Comment (jakermx) [reply resources/language/resource.language.en_gb/strings.po]: Haven´t heard about it, but googling it, it exists...so 640p its ok

File: resources/language/resource.language.en_gb/strings.po (hunk 8 above)
Comment (jakermx) [reply resources/language/resource.language.en_gb/strings.po]: 480p its ok

File: resources/language/resource.language.en_gb/strings.po (hunk 9 above)
Comment (jakermx) [reply resources/language/resource.language.en_gb/strings.po]: I think we need to be standard, so if we use SD/HD res_h"p" its oks,

File: resources/language/resource.language.en_gb/strings.po (hunk 10 above)
Comment (jakermx) [reply resources/language/resource.language.en_gb/strings.po]: I think we need to be standard, so if we use SD/HD res_h"p" its oks,

**Comment by jakermx** [comment]
> Also, a question. What if I don't want to limit the resolution at all and have IE 8K whan avaliable without waiting for an update? How can I do? If I'm correct, your setting is mandatory, and ther's no way to avoid limiting...
> 
> Last but not least, for me, SD should limit to 480p, not 639.

You can set it to 4k ,it should incluide all, unless we add an Resolution Limit  flag thru settings

File: resources/lib/services/msl/profiles.py (hunk 6 above)
Comment (Smeulf) [reply resources/lib/services/msl/profiles.py]: You can use common.debug() instead ;)

File: resources/settings.xml (hunk 11 above)
Comment (jakermx) [reply resources/settings.xml]: I think we should set it to max...so 4k, or I can add a Max Option with 9999 value

File: resources/settings.xml (hunk 1 above)
Comment (jakermx) [reply resources/settings.xml]: It wasnt, I removed some empty lines...not sure, but let me check it

**Comment by Smeulf** [comment]
> > Also, a question. What if I don't want to limit the resolution at all and have IE 8K whan avaliable without waiting for an update? How can I do? If I'm correct, your setting is mandatory, and ther's no way to avoid limiting...
> > Last but not least, for me, SD should limit to 480p, not 639.
> 
> You can set it to 4k ,it should incluide all, unless we add an Resolution Limit flag thru settings

So what about call it 'Max' or 'No limit'?

**Comment by jakermx** [comment]
> > > Also, a question. What if I don't want to limit the resolution at all and have IE 8K whan avaliable without waiting for an update? How can I do? If I'm correct, your setting is mandatory, and ther's no way to avoid limiting...
> > > Last but not least, for me, SD should limit to 480p, not 639.
> > 
> > 
> > You can set it to 4k ,it should incluide all, unless we add an Resolution Limit flag thru settings
> 
> So what about call it 'Max' or 'No limit'?

Agreed, No limit sounds good to me....

**Comment by jakermx** [comment]
I will try this on Pi Zero W, maybe setting it to Min Available works.... 

:)

**Comment by Smeulf** [comment]
> Agreed, No limit sounds good to me....

Easier to understand ;)

**Comment by Smeulf** [comment]
> I will try this on Pi Zero W, maybe setting it to Min Available works....
> 
> :)

No way!! lol 
Buy a Pi4 when temperature and USB-C issues are fixed :p

**Comment by jakermx** [comment]
> > I will try this on Pi Zero W, maybe setting it to Min Available works....
> > :)
> 
> No way!! lol
> Buy a Pi4 when temperature and USB-C issues are fixed :p

I have 2 and 3 both b+, I am waiting pi 4 with 4GB to be available... but I got a new NVIDIA Jetson Nano...I will try it when It arrives

**Comment by CastagnaIT** [comment]
where did you get the profiles and levels you added?
I don't see them inside the netflix playercore

**Comment by jakermx** [comment]
Hello,

I read that someone on asciidisco were complaing  about Limits on InputStreams affects other AddOns, so I started looking for a codec limiting solution, so first I read a little abot codecs, profiles and levels, then I thought that there are somewhere in intenet a manifest that would restrict profiles, because we did the MSLv2 develpment when Netflix required VP9 profiles,so then I went to pywidevine so I had very hard time loking for the source code, and finally I ot it from a cache server, and I got a Mega link for a completehacking framework that is about 2 GB, anyway, it contains the pywv client that has these:

PROFILES = {
    'h264': {
        'sd': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264hpl31-dash',
        ],
        '720p': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264bpl31-dash',
            'playready-h264mpl31-dash',
			'playready-h264hpl31-dash',
        ],
        '1080p': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264bpl31-dash',
            'playready-h264mpl31-dash',
			'playready-h264hpl31-dash',
			'playready-h264bpl40-dash',
            'playready-h264mpl40-dash',
			'playready-h264hpl40-dash',
        ]
    },
    'hevc': {
        'sd': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264hpl31-dash',
			'hevc-main-L30-dash-cenc',
			'hevc-main10-L30-dash-cenc',
			'hevc-main10-L30-dash-cenc-prk',			
			'hevc-main-L30-L31-dash-cenc-tl',
			'hevc-main10-L30-L31-dash-cenc-tl',
        ],
        '720p': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264bpl31-dash',
            'playready-h264mpl31-dash',
			'playready-h264hpl31-dash',
			'hevc-main-L30-dash-cenc',
			'hevc-main10-L30-dash-cenc',
			'hevc-main10-L30-dash-cenc-prk',
			'hevc-main-L31-dash-cenc',
			'hevc-main10-L31-dash-cenc',
			'hevc-main10-L31-dash-cenc-prk',			
			'hevc-main-L30-L31-dash-cenc-tl',
			'hevc-main10-L30-L31-dash-cenc-tl',			
        ],
        '1080p': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264bpl31-dash',
            'playready-h264mpl31-dash',
			'playready-h264hpl31-dash',
			'playready-h264bpl40-dash',
            'playready-h264mpl40-dash',
			'playready-h264hpl40-dash',
			'hevc-main-L30-dash-cenc',
			'hevc-main10-L30-dash-cenc',
			'hevc-main10-L30-dash-cenc-prk',
			'hevc-main-L31-dash-cenc',
			'hevc-main10-L31-dash-cenc',
			'hevc-main10-L31-dash-cenc-prk',
			'hevc-main-L40-dash-cenc',
			'hevc-main10-L40-dash-cenc',
			'hevc-main10-L40-dash-cenc-prk',			
			'hevc-main-L31-L40-dash-cenc-tl',
			'hevc-main10-L31-L40-dash-cenc-tl',	
        ]
    },
    'hdr': {
        'sd': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264hpl31-dash',
			'hevc-hdr-main-L30-dash-cenc',
			'hevc-hdr-main10-L30-dash-cenc',
			'hevc-hdr-main10-L30-dash-cenc-prk',			
			'hevc-hdr-main-L30-L31-dash-cenc-tl',
			'hevc-hdr-main10-L30-L31-dash-cenc-tl',
        ],
        '720p': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264bpl31-dash',
            'playready-h264mpl31-dash',
			'playready-h264hpl31-dash',
			'hevc-hdr-main-L30-dash-cenc',
			'hevc-hdr-main10-L30-dash-cenc',
			'hevc-hdr-main10-L30-dash-cenc-prk',
			'hevc-hdr-main-L31-dash-cenc',
			'hevc-hdr-main10-L31-dash-cenc',
			'hevc-hdr-main10-L31-dash-cenc-prk',			
			'hevc-hdr-main-L30-L31-dash-cenc-tl',
			'hevc-hdr-main10-L30-L31-dash-cenc-tl',			
        ],
        '1080p': [
			'playready-h264bpl30-dash',
			'playready-h264hpl30-dash',
            'playready-h264mpl30-dash',
			'playready-h264bpl31-dash',
            'playready-h264mpl31-dash',
			'playready-h264hpl31-dash',
			'playready-h264bpl40-dash',
            'playready-h264mpl40-dash',
			'playready-h264hpl40-dash',
			'hevc-hdr-main-L30-dash-cenc',
			'hevc-hdr-main10-L30-dash-cenc',
			'hevc-hdr-main10-L30-dash-cenc-prk',
			'hevc-hdr-main-L31-dash-cenc',
			'hevc-hdr-main10-L31-dash-cenc',
			'hevc-hdr-main10-L31-dash-cenc-prk',
			'hevc-hdr-main-L40-dash-cenc',
			'hevc-hdr-main10-L40-dash-cenc',
			'hevc-hdr-main10-L40-dash-cenc-prk',			
			'hevc-hdr-main-L31-L40-dash-cenc-tl',
			'hevc-hdr-main10-L31-L40-dash-cenc-tl',			
        ]
    },
    'audio': [
        #'heaac-2-dash',
        'ddplus-2.0-dash',
        'ddplus-5.1hq-dash',
        'ddplus-atmos-dash',
        'dd-5.1-dash',
    ],
    'subs': [
        'dfxp-ls-sdh',
        'simplesdh',
        #'nflx-cmisc',
        #'webvtt-lssdh-ios8',
        #'webvtt-lssdh-ios'
        #'BIF240',
        #'BIF320',
    ]


so I did a resolution support per profile matrix, and i did what I sent first... but dont know whyy Netflix requieres at least having one h264 Level 3.1 on he manifest, so I change my mind and did what you are seeing on this PR.

VP9 Profiles0 and Pofiles2 I got them from a "Official" Java Client source code from Netflx that were never released but forked by few ones... basically, it ask a for all combinations with profile0 and profile2, with all the available Levels, that suppost to have 8k support....

I did not think it was  wrong, since he AddOn just propose some profiles and Netflix responds with what i thinks is better, i.e. If you go to a fried house and her mother ask you if you would like water, coke, beer or wine... you will chooose what ever you think its better for you at that moment....so that is why I added that many profiles.

but the selections occurs aftter receiving a proposal...removing from MPD creation process higher than it was set.

Salud!!!

**Comment by jakermx** [comment]
FYI Cadmium player is just an free option, it does not includes all the functionallity available from the framework, like 1080p extension that release it.

and BTW, I am doing same procedure to find out a way to set the watched flag...but it will be more complex, because it starts requesting for a track_id on the model request proccess that would be used be pbo_Events manifest to set that flag within other stuff.

**Comment by Smeulf** [comment]
> but dont know whyy Netflix requieres at least having one h264 Level 3.1 on he manifest

To ensure they can deliver at least one compatible stream I presume

**Comment by CastagnaIT** [comment]
interesting but even if you found the widewine sources, it doesn't mean that the netflix services use everything you see there

**Comment by CastagnaIT** [comment]
I did some tests with the new profiles you added,
i never got any results, so i don't think they are used by the netflix backend

if you find a match that your new profiles works, mention the manifest and the title of video

for now i can only implement your request to impose the limit by modifying the manifest

**Comment by quickstreamingmedia** [comment]
> Hello,
> 
> I read that someone on asciidisco were complaing about Limits on InputStreams affects other AddOns, so I started looking for a codec limiting solution, so first I read a little abot codecs, profiles and levels, then I thought that there are somewhere in intenet a manifest that would restrict profiles, because we did the MSLv2 develpment when Netflix required VP9 profiles,so then I went to pywidevine so I had very hard time loking for the source code, and finally I ot it from a cache server, and I got a Mega link for a completehacking framework that is about 2 GB, anyway, it contains the pywv client that has these:
> 
> PROFILES = {
> 'h264': {
> 'sd': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264hpl31-dash',
> ],
> '720p': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264bpl31-dash',
> 'playready-h264mpl31-dash',
> 'playready-h264hpl31-dash',
> ],
> '1080p': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264bpl31-dash',
> 'playready-h264mpl31-dash',
> 'playready-h264hpl31-dash',
> 'playready-h264bpl40-dash',
> 'playready-h264mpl40-dash',
> 'playready-h264hpl40-dash',
> ]
> },
> 'hevc': {
> 'sd': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264hpl31-dash',
> 'hevc-main-L30-dash-cenc',
> 'hevc-main10-L30-dash-cenc',
> 'hevc-main10-L30-dash-cenc-prk',
> 'hevc-main-L30-L31-dash-cenc-tl',
> 'hevc-main10-L30-L31-dash-cenc-tl',
> ],
> '720p': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264bpl31-dash',
> 'playready-h264mpl31-dash',
> 'playready-h264hpl31-dash',
> 'hevc-main-L30-dash-cenc',
> 'hevc-main10-L30-dash-cenc',
> 'hevc-main10-L30-dash-cenc-prk',
> 'hevc-main-L31-dash-cenc',
> 'hevc-main10-L31-dash-cenc',
> 'hevc-main10-L31-dash-cenc-prk',
> 'hevc-main-L30-L31-dash-cenc-tl',
> 'hevc-main10-L30-L31-dash-cenc-tl',
> ],
> '1080p': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264bpl31-dash',
> 'playready-h264mpl31-dash',
> 'playready-h264hpl31-dash',
> 'playready-h264bpl40-dash',
> 'playready-h264mpl40-dash',
> 'playready-h264hpl40-dash',
> 'hevc-main-L30-dash-cenc',
> 'hevc-main10-L30-dash-cenc',
> 'hevc-main10-L30-dash-cenc-prk',
> 'hevc-main-L31-dash-cenc',
> 'hevc-main10-L31-dash-cenc',
> 'hevc-main10-L31-dash-cenc-prk',
> 'hevc-main-L40-dash-cenc',
> 'hevc-main10-L40-dash-cenc',
> 'hevc-main10-L40-dash-cenc-prk',
> 'hevc-main-L31-L40-dash-cenc-tl',
> 'hevc-main10-L31-L40-dash-cenc-tl',
> ]
> },
> 'hdr': {
> 'sd': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264hpl31-dash',
> 'hevc-hdr-main-L30-dash-cenc',
> 'hevc-hdr-main10-L30-dash-cenc',
> 'hevc-hdr-main10-L30-dash-cenc-prk',
> 'hevc-hdr-main-L30-L31-dash-cenc-tl',
> 'hevc-hdr-main10-L30-L31-dash-cenc-tl',
> ],
> '720p': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264bpl31-dash',
> 'playready-h264mpl31-dash',
> 'playready-h264hpl31-dash',
> 'hevc-hdr-main-L30-dash-cenc',
> 'hevc-hdr-main10-L30-dash-cenc',
> 'hevc-hdr-main10-L30-dash-cenc-prk',
> 'hevc-hdr-main-L31-dash-cenc',
> 'hevc-hdr-main10-L31-dash-cenc',
> 'hevc-hdr-main10-L31-dash-cenc-prk',
> 'hevc-hdr-main-L30-L31-dash-cenc-tl',
> 'hevc-hdr-main10-L30-L31-dash-cenc-tl',
> ],
> '1080p': [
> 'playready-h264bpl30-dash',
> 'playready-h264hpl30-dash',
> 'playready-h264mpl30-dash',
> 'playready-h264bpl31-dash',
> 'playready-h264mpl31-dash',
> 'playready-h264hpl31-dash',
> 'playready-h264bpl40-dash',
> 'playready-h264mpl40-dash',
> 'playready-h264hpl40-dash',
> 'hevc-hdr-main-L30-dash-cenc',
> 'hevc-hdr-main10-L30-dash-cenc',
> 'hevc-hdr-main10-L30-dash-cenc-prk',
> 'hevc-hdr-main-L31-dash-cenc',
> 'hevc-hdr-main10-L31-dash-cenc',
> 'hevc-hdr-main10-L31-dash-cenc-prk',
> 'hevc-hdr-main-L40-dash-cenc',
> 'hevc-hdr-main10-L40-dash-cenc',
> 'hevc-hdr-main10-L40-dash-cenc-prk',
> 'hevc-hdr-main-L31-L40-dash-cenc-tl',
> 'hevc-hdr-main10-L31-L40-dash-cenc-tl',
> ]
> },
> 'audio': [
> #'heaac-2-dash',
> 'ddplus-2.0-dash',
> 'ddplus-5.1hq-dash',
> 'ddplus-atmos-dash',
> 'dd-5.1-dash',
> ],
> 'subs': [
> 'dfxp-ls-sdh',
> 'simplesdh',
> #'nflx-cmisc',
> #'webvtt-lssdh-ios8',
> #'webvtt-lssdh-ios'
> #'BIF240',
> #'BIF320',
> ]
> 
> so I did a resolution support per profile matrix, and i did what I sent first... but dont know whyy Netflix requieres at least having one h264 Level 3.1 on he manifest, so I change my mind and did what you are seeing on this PR.
> 
> VP9 Profiles0 and Pofiles2 I got them from a "Official" Java Client source code from Netflx that were never released but forked by few ones... basically, it ask a for all combinations with profile0 and profile2, with all the available Levels, that suppost to have 8k support....
> 
> I did not think it was wrong, since he AddOn just propose some profiles and Netflix responds with what i thinks is better, i.e. If you go to a fried house and her mother ask you if you would like water, coke, beer or wine... you will chooose what ever you think its better for you at that moment....so that is why I added that many profiles.
> 
> but the selections occurs aftter receiving a proposal...removing from MPD creation process higher than it was set.
> 
> Salud!!!

pywidevine do you got this if yes contact me on telegram @alienmaster9

**Pull request closed without merging**